### PR TITLE
CR for #20422 - Add the ability to manage friends via Interface

### PR DESCRIPTION
**Pull request by ctrlaltdavid**

### CODE REVIEW

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -4013,6 +4017,32 @@ void Application::loadScriptURLDialog() {
...
+        connect(_editFriendsDialog, &QWidget::destroyed, this, &Application::editFriendsDialogDestroyed);
+    }
+
+    QMetaObject::invokeMethod(_editFriendsDialog, "setVisible", Qt::AutoConnection, Q_ARG(bool, true));
+    QMetaObject::invokeMethod(_editFriendsDialog, "showNormal", Qt::AutoConnection);
+}
+
+void Application::editFriendsDialogDestroyed() {
+    _editFriendsDialog = nullptr;
+}
```
Comment (birarda) [inline interface/src/Application.cpp]: You should be able to do this with the WebWindow class from JS? See how Edit.js does it for the marketplace.

**Comment by hifi-gustavo** [comment]
Build console output for commit: e06f28b7c5c74d796514a99797491a07899dae76 available at: https://gist.github.com/e1175ba8df6d3b51e8dd<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4565.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4565.dmg

File: interface/src/Application.cpp (hunk 1 above)
Comment (ctrlaltdavid) [inline interface/src/Application.cpp]: Yes, I used WebWindows for starters when opening the friends window from JavaScript, but it comes with a bunch of ScriptEngine baggage that isn't good for when opening the friends window from a menu item. ... Perhaps WebWindow should derive from a simpler class such as this in due course.

**Comment by hifi-gustavo** [comment]
Build console output for commit: e51992832047af6f1749a14db11b2caa34143118 available at: https://gist.github.com/e1175ba8df6d3b51e8dd<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4565.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4565.dmg

File: interface/src/Application.cpp (hunk 1 above)
Comment (ZappoMan) [inline interface/src/Application.cpp]: I don't understand what you're saying here David... I am using WebWindow from the preferences dialog box and it doesn't seem to be a problem.

Comment (ctrlaltdavid) [inline interface/src/Application.cpp]: @ZappoMan I can't see any mention of WebWindow in PreferencesDialog? ... The friends window is now implemented in its own class, FriendsWindow.

**Comment by birarda** [comment]
@ctrlaltdavid if you check HEAD you can see that the prefs dialog displays a WebWindow for marketplace avatars, shouldn't we be able to do the same without creating a whole new widget?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 349cc3241da2eedc0a43770088cb203fb8418bc1 available at: https://gist.github.com/e1175ba8df6d3b51e8dd<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4565.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4565.dmg

File: interface/src/Menu.cpp [hunk 2]
```diff
@@ -152,6 +152,32 @@ Menu::Menu() {
     
     addActionToQMenuAndActionHash(toolsMenu, MenuOption::Chat, Qt::Key_Backslash,
                                   dialogsManager.data(), SLOT(showIRCLink()));
+    addActionToQMenuAndActionHash(toolsMenu, MenuOption::AddRemoveFriends, 0,
+                                  qApp, SLOT(showFriendsWindow()));
+
+    QMenu* visibilityMenu = toolsMenu->addMenu("I Am Visible To");
```
Comment (jherico) [inline interface/src/Menu.cpp]: If I change this setting from the Qt menu, the change isn't reflected in the in-world user panel.  However if I change it in the in-world panel, it IS reflected in Qt menu.  The visibility status should be bidirectional, or only visible / changable in one place (but prefer the former).

Comment (ctrlaltdavid) [inline interface/src/Menu.cpp]: @jherico You need to run the updated users.js contained in this PR; it does exactly that which you suggest.

Comment (jherico) [inline interface/src/Menu.cpp]: so it does.

**Pull request merged**